packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -915,6 +915,11 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
 
   @autobind
   private _onSubMenuDismiss(ev?: any, dismissAll?: boolean) {
```

Comment:
A potential alternative, which I would prefer to adding another property, is to have _onSubMenuDismiss take in item as an argument. That way might look something like this: 

`` 
private _onSubMenuDismiss(ev?: any, dismissAll?: boolean, item?: IContextualMenuItem) {
    if(item && item.subMenuProps && item.subMenuProps && item.SubMenuProps.onDismiss) {
         item.SubMenuProps.onDismiss(ev, dismissAll);
    }
``

Then when you're creating the submenu props in _getSubmenuProps you would add

``
     if (item) {
       submenuProps = {
         items: getSubmenuItems(item)!,
         target: submenuTarget,
         onDismiss: (ev: any, dismissAll: boolean) => this._onSubMenuDismiss(ev, dismissAll, item),
         isSubMenu: true,
         id: this.state.subMenuId,
         shouldFocusOnMount: true,
         directionalHint: getRTL() ? DirectionalHint.leftTopEdge : DirectionalHint.rightTopEdge,
         className: this.props.className,
         gapSpace: 0,
         isBeakVisible: false
       };
``
Or you could add a function that curries the item into this._onSubmenuDismiss
``
private _getDismissSubmenu(item) {
    return (ev: any, dismissAll: boolean) => this._onSubmenuDismiss(ev, dismissAll, item);
}
``

And then the call is much cleaner in getSubmenuProps.

Reply:
Wow. That's so much cleaner than what I want. Really great suggestion, I'm really, really impressed with this refactor that you suggested! Thanks! I'll do this.